### PR TITLE
feat: Try to Support flavors (android & ios)

### PR DESCRIPTION
**Pull request by salahamassi**
## Description

Firebase CLI will try to get all flavors for the android project and variant build configrations for the ios project 
and show a prompt to the ser 

`? Do you want  to create a firebase project for each flavor? (y/n)
`

if the user selects "y" for each flavor firebase CLI will create a firebase project with selected platforms

- create `firebase_options.dart` for each flavor
  For example:
  (lib/firebase_options_development.dart,
  lib/firebase_options_production.dart,
  lib/firebase_options_staging.dart)

- create `firebase_app_id_file.json` for each flavor
  For example:
  (ios/firebase_app_id_file_development.json,
  ios/firebase_app_id_file_production.json,
   ios/firebase_app_id_file_staging.json)

- create `google-services.json`  for each flavor 
  For example:
  (android/app/src/development/google-services.json,
   android/app/src/production/google-services.json,
   android/app/src/staging/google-services.json)

- create `GoogleService-Info.plist` for each flavor 
  For example:
  (ios/Runner/development/GoogleService-Info.plist,
   ios/Runner/production/GoogleService-Info.plist,
   ios/Runner/staging/GoogleService-Info.plist)

For ios Project add shell script to copy the correct  `GoogleService-Info.plist` to the `BUILT_PRODUCTS_DIR` based on the selected ios schema:
```shell
GOOGLE_SERVICE_INFO_PLIST_FROM="${PROJECT_DIR}/Runner/${FLAVOR}/GoogleService-Info.plist"
BUILD_APP_DIR="${BUILT_PRODUCTS_DIR}/${FULL_PRODUCT_NAME}"
GOOGLE_SERVICE_INFO_PLIST_TO="${BUILT_PRODUCTS_DIR}/${PRODUCT_NAME}.app"
cp "${GOOGLE_SERVICE_INFO_PLIST_FROM}" "${GOOGLE_SERVICE_INFO_PLIST_TO}"
```
 
Please try it 🙏🚀

## Type of Change

- [x] ✨ `feat` -- New feature (non-breaking change which adds functionality)
- [ ] 🛠️ `fix` -- Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ `!` -- Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 `refactor` -- Code refactor
- [ ] ✅ `ci` -- Build configuration change
- [ ] 📝 `docs` -- Documentation
- [ ] 🗑️ `chore` -- Chore

### CODE REVIEW

**Comment by salahamassi** [comment]
I opened this PR based on this issue 
https://github.com/invertase/flutterfire_cli/issues/14

**Comment by Lyokone** [comment]
I've try the PR using the VGV CLI: 
```
very_good create project --desc "Project" --org "com.lyokone.project" 
```

It produces the following result:
```
FileSystemException: Cannot open file, path = 'project/android/app/src/production/google-services.json' (OS Error: No such file or directory, errno = 2)
```

I feel like matching the name of the flavor with a folder can produce unexpected results on Android 🤔

**Comment by salahamassi** [comment]
> I've try the PR using the VGV CLI:
> 
> ```
> very_good create project --desc "Project" --org "com.lyokone.project" 
> ```
> 
> It produces the following result:
> 
> ```
> FileSystemException: Cannot open file, path = 'project/android/app/src/production/google-services.json' (OS Error: No such file or directory, errno = 2)
> ```
> 
> I feel like matching the name of the flavor with a folder can produce unexpected results on Android 🤔

I think Each flavor must have a folder with its name

I tried to fix this by checking if the folder exists before creating `androidGoogleServicesJsonFile`
any suggestions?

File: packages/flutterfire_cli/lib/src/commands/config.dart [hunk 1]
```diff
@@ -365,7 +369,67 @@ class ConfigCommand extends FlutterFireCommand {
   Future<void> run() async {
     commandRequiresFlutterApp();
 
-    final selectedFirebaseProject = await _selectFirebaseProject();
+    var supportFlavors = false;
+    if (flutterApp!.hasFlavors) {
+      supportFlavors = promptBool(
+        'Do you want  to create a firebase project for each flavor?',
```
Comment (Salakar) [inline packages/flutterfire_cli/lib/src/commands/config.dart:375]:
```suggestion
        'Do you want to create a firebase project for each flavor?',
```

**Comment by ulisseshen** [comment]
any news about this PR?

**Comment by Lyokone** [comment]
@ulisseshen, @russellwheatley is currently working on this feature in this PR: https://github.com/invertase/flutterfire_cli/pull/132

**Comment by ulisseshen** [comment]
> @ulisseshen, @russellwheatley is currently working on this feature in this PR: #132

thanks for response.

**Comment by russellwheatley** [comment]
Closing as this is no longer necessary. See https://github.com/invertase/flutterfire_cli/pull/132

**Pull request closed without merging**